ctapipe/reco/hillas_intersection.py

```diff
@@ -0,0 +1,405 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+
+TODO:
+ - Speed tests, need to be certain the looping on all telescopes is not killing 
+performance
+- Introduce new weighting schemes
+- Make intersect_lines code more readable
+
+"""
+import numpy as np
+import itertools
+import astropy.units as u
+from ctapipe.reco.reco_algorithms import Reconstructor
+from ctapipe.io.containers import ReconstructedShowerContainer
+from ctapipe.coordinates import *
+from ctapipe.instrument import get_atmosphere_profile_functions
+
+__all__ = [
+    'HillasIntersection'
+]
+
+
+class HillasIntersection(Reconstructor):
+    '''
+    This class is a simple re-implementation of Hillas parameter based event 
+    reconstruction.
+    e.g. https://arxiv.org/abs/astro-ph/0607333
+
+    In this case the Hillas parameters are all constructed in the shared angular (
+    Nominal) system. Direction reconstruction is performed by extrapolation of the 
+    major axes of the Hillas parameters in the nominal system and the weighted average 
+    of the crossing points is taken. Core reconstruction is performed by performing the 
+    same procedure in the tilted ground system.
+
+    The height of maximum is reconstructed by the projection os the image centroid onto 
+    the shower axis, taking the weighted average of all images.
+
+    Uncertainties on the positions are provided by taking the spread of the crossing 
+    points, however this means that no uncertainty can be provided for multiplicity 2 
+    events.
+
+    '''
+
+    def __init__(self, atmosphere_profile_name="paranal"):
+
+        # We need a conversion function from height above ground to depth of maximum
+        # To do this we need the conversion table from CORSIKA
+        self.thickness_profile, self.altitude_profile = get_atmosphere_profile_functions(
+            atmosphere_profile_name)
+
+    def predict(self, hillas_parameters, tel_x, tel_y, array_direction):
```

Comment:
Should we try to get the interface of the various `predict` functions as uniform as possible?
for comparison:
`HillasReconstructor.predict(self, hillas_dict, inst, tel_phi, tel_theta, seed_pos=(0, 0))`
`ImPACTReconstructor.predict(self, shower_seed, energy_seed)`

The telescope positions are in the `inst` object, my `tel_phi` and `tel_theta` should go in there as well...
The `seed_pos` isn't even used any more but we could add some default `XXX_seed=None` everywhere.
This would help switching out the reconstruction method on the fly without having to think too much about the function call itself. Or do you think we should just add `*args, **kwargs` at the end of the method ;)

Just a though.